openbsd/OpenBSDMachine.c

```diff
@@ -105,8 +106,9 @@ Machine* Machine_new(UsersTable* usersTable, uid_t userId) {
       CRT_fatalError("fscale sysctl call failed");
    }
 
-   if ((this->pageSize = sysconf(_SC_PAGESIZE)) == -1)
+   if ((pageSize = sysconf(_SC_PAGESIZE)) == -1)
       CRT_fatalError("pagesize sysconf call failed");
+   this->pageSize = (size_t) pageSize;
```

Comment:
Converting `pageSize` into `size_t` helped resolve this.

```
  openbsd/OpenBSDMachine.c:108:26: error: implicit conversion loses integer precision: 'long' to 'int' [-Werror,-Wshorten-64-to-32]
    108 |    if ((this->pageSize = sysconf(_SC_PAGESIZE)) == -1)
        |                        ~ ^~~~~~~~~~~~~~~~~~~~~
  1 error generated.
```

Reply:
Move the `pageSize` declaration to just before the `sysconf` call. And if we're casting it to `size_t`, filter out all negative values.

```c
   long pageSize = sysconf(_SC_PAGESIZE);
   if (pageSize < 0)
      CRT_fatalError("pagesize sysconf call failed");
   this->pageSize = (size_t) pageSize;
```

Reply:
I re-proposed this in #1827.
(No, I don't think the one-liner is good for what I was trying to solve. My aim is to filter out all invalid values and make the signed `long` to unsigned `size_t` cast safe.)